### DISCUSSION TITLE
Add sampling to the GPX trace import

### DISCUSSION THREAD

**Discussion by glouvigny** [Freika/dawarich#292] [Ideas]
Hi there,

I'm currently playing with workout traces exported from my iPhone, which you can get via the export function of the Apple Health app.

It provides traces in the GPX format that can be properly imported into dawarich. The traces contain a point per second, and the coordinates have a precision of 6 decimal places.

While it's great to have that much data, it definitely hinders the performance of the app.

In total, I have 1,391,944 points for 929 traces over a period of 2 years, and some pages like the yearly map won't load.

Reverse geocoding on these points also takes a lot of time.

Is it possible to add a sampling option for imported traces (e.g., keep only x points per minute), or alternatively reduce the coordinate precision?

**Comment by Freika** [comment]
It totally makes sense and I'll implement it at some point. Thanks for suggestion!

**Comment by Freika** [comment]
@glouvigny release [0.15.8](https://github.com/Freika/dawarich/releases/tag/0.15.8) introduces "Simplified" mode of points rendering. When enabled, points that are closer to each other than 50 meters or 20 seconds, will not be rendered, improving the map performance. Let me know if you'll encounter any issues with that
